src/compiler/checker.ts

```diff
@@ -28040,26 +28040,31 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         }
 
         function narrowTypeBySwitchOnTrue(type: Type, switchStatement: SwitchStatement, clauseStart: number, clauseEnd: number): Type {
-            const clauses = switchStatement.caseBlock.clauses.slice(clauseStart, clauseEnd);
-            const clausesType = narrowTypeForTrueClauses(type, clauses);
             const defaultIndex = findIndex(switchStatement.caseBlock.clauses, clause => clause.kind === SyntaxKind.DefaultClause);
             const hasDefaultClause = clauseStart === clauseEnd || (defaultIndex >= clauseStart && defaultIndex < clauseEnd);

```

Comment:
Reading this in the below code was confusing for me because it's not about the `switch` having a default, it's about the given range including a default.

Reply:
This is just the terminology used above by the `typeof` narrowing. I can change it, but it will be inconsistent.